python/cuml/neighbors/nearest_neighbors.pyx

```diff
@@ -327,22 +243,13 @@ cdef class NearestNeighbors:
                        <int> 1)
 
     def _fit_mg(self, n_dims, alloc_info):
-        """
-        Fits a model using multiple GPUs. This method takes in a list of dict objects
-        representing the distribution of the underlying device pointers. The device
-        information can be extracted from the pointers.
 
-        :param n_dims
-            the number of features for each vector
-        :param alloc_info
-            a list of __cuda_array_interface__ dicts
-        :return:
-        """
 
         if self.k != NULL:
             del self.k
 
-        self.k = new kNN(n_dims, verbose = self._verbose)
+        cdef cumlHandle * handle_ = < cumlHandle * > < size_t > self.handle.getHandle()
+        self.k = new kNN(handle_[0], n_dims, verbose = self._verbose)
```

Comment:
Missing handle_.sync() after this

Reply:
Maybe I'm missing the point of cumlHandle in the cython layer, but the kNN constructor is a blocking operation. Why would a sync be needed here?

Reply:
Ah true, was thinking of general fit methods/transform methods mentioned above, which I believe to still apply the missing sync, no?